src/blocks-ext.js

```diff
@@ -444,42 +466,55 @@ HintInputSlotMorph.prototype = new InputSlotMorph();
 HintInputSlotMorph.prototype.constructor = HintInputSlotMorph;
 HintInputSlotMorph.uber = InputSlotMorph.prototype;
 
-function HintInputSlotMorph(text, hint, isNumeric) {
+function HintInputSlotMorph(text, hint, isNumeric, choiceDict, isReadOnly) {
     var self = this;
 
     this.hintText = hint;
     this.empty = true;
-    InputSlotMorph.call(this, text, isNumeric);
+    InputSlotMorph.call(this, text, isNumeric, choiceDict, isReadOnly);
 
-    // If the StringMorph gets clicked on when empty, the hint text
-    // should be "ghostly"
-    this.contents().mouseClickLeft = function() {
-        if (self.isEmptySlot()) {
-            this.text = '';
-        }
-        StringMorph.prototype.mouseClickLeft.apply(this, arguments);
-    };
+    // If the StringMorph gets clicked on when empty, the hint text should be "ghostly"
+    if (!choiceDict) { // don't wipe out enum inputs
+        this.contents().mouseClickLeft = function() {
+            if (self.isEmptySlot()) {
+                this.text = '';
+            }
+            StringMorph.prototype.mouseClickLeft.apply(this, arguments);
+        };
+    }
 }
 
 HintInputSlotMorph.prototype.evaluate = function() {
     if (this.isEmptySlot()) {  // ignore grey text
-        return this.isNumeric ? 0 : '';
+        return '';
     }
     return InputSlotMorph.prototype.evaluate.call(this);
 };
 
+const WHITE_HINT = new Color(190, 190, 190);
+const BLACK_HINT = new Color(100, 100, 100);
 HintInputSlotMorph.prototype.setContents = function(value) {
-    var color = BLACK,
-        contents = this.contents();
+    const contents = this.contents();
+    let color;
 
-    // If empty, set to the hint text
     InputSlotMorph.prototype.setContents.apply(this, arguments);
-    this.empty = value === '';
-    if (this.isEmptySlot()) {  // Set the contents to the hint text
-        // Set the text to the hint text
+
+    this.overrideWhite = null;
```

Comment:
Rather than optionally setting these attributes and short circuiting every time we use them, it might be nice to just set the to `WHITE` and `BLACK` by default and use them every time. I would also recommend renaming them to something like `lightColor` and `darkColor` (or perhaps `lightTextColor` and `darkTextColor`)

Reply:
That should work for most places, but it'd need new fields for `BlockLabelMorph`, `BlockSymbolMorph`, and `BlockMorph`, and there are 2 places that access parent/children, so those would still need short circuiting in case it's not one of those types. Either way we're making a change from base snap, so I guess it's just whichever seems cleaner.

Reply:
Gotcha. In that case, it is probably fine.